test/dotnet-sln.Tests/GivenDotnetSlnAdd.cs

```diff
@@ -1346,108 +1058,150 @@ public void WhenSolutionFolderAndInRootIsPassedItFails(string solutionCommand)
+                .ToList();
             if (matchingSolutionFolder.Count == 1)
             {
-                slnContents = slnContents.Replace("__SOLUTION_FOLDER_GUID__", matchingSolutionFolder[0].Id);
+                slnContents = slnContents.Replace(
+                    "_SOLUTION_FOLDER_GUID__",
+                    solutionExtension == ".sln"
+                        ? $"{{{matchingSolutionFolder[0].Id.ToString().ToUpper()}}}"
+                        : matchingSolutionFolder[0].Id.ToString());
             }
 
             return slnContents;
         }
 
         [Theory]
-        [InlineData("sln")]
-        [InlineData("solution")]
-        public void WhenSolutionIsPassedAsProjectItPrintsSuggestionAndUsage(string solutionCommand)
+        [InlineData("sln", ".sln")]
+        [InlineData("solution", ".sln")]
+        public void WhenSolutionIsPassedAsProjectItPrintsSuggestionAndUsage(string solutionCommand, string solutionExtension)
         {
-            VerifySuggestionAndUsage(solutionCommand, "");
+            VerifySuggestionAndUsage(solutionCommand, "", solutionExtension);
         }
 
         [Theory]
-        [InlineData("sln")]
-        [InlineData("solution")]
-        public void WhenSolutionIsPassedAsProjectWithInRootItPrintsSuggestionAndUsage(string solutionCommand)
+        [InlineData("sln", ".sln")]
+        [InlineData("solution", ".sln")]
+        public void WhenSolutionIsPassedAsProjectWithInRootItPrintsSuggestionAndUsage(string solutionCommand, string solutionExtension)
         {
-            VerifySuggestionAndUsage(solutionCommand, "--in-root");
+            VerifySuggestionAndUsage(solutionCommand, "--in-root", solutionExtension);
         }
 
         [Theory]
-        [InlineData("sln")]
-        [InlineData("solution")]
-        public void WhenSolutionIsPassedAsProjectWithSolutionFolderItPrintsSuggestionAndUsage(string solutionCommand)
+        [InlineData("sln", ".sln")]
+        [InlineData("solution", ".sln")]
+        public void WhenSolutionIsPassedAsProjectWithSolutionFolderItPrintsSuggestionAndUsage(string solutionCommand, string solutionExtension)
         {
-            VerifySuggestionAndUsage(solutionCommand, "--solution-folder");
+            VerifySuggestionAndUsage(solutionCommand, "--solution-folder", solutionExtension);
         }
-        private void VerifySuggestionAndUsage(string solutionCommand, string arguments)
+        private void VerifySuggestionAndUsage(string solutionCommand, string arguments, string solutionExtension)
         {
             var projectDirectory = _testAssetsManager
-                .CopyTestAsset("TestAppWithSlnAndCsprojFiles", identifier: $"{solutionCommand}{arguments}")
+                .CopyTestAsset("TestAppWithSlnAndCsprojFiles", identifier: $"{solutionCommand}{arguments}{solutionExtension}")
                 .WithSource()
                 .Path;
 
+            // TODO: Move to different location
```

Comment:
nit: do we want to do this? I dont think it is needed.

Reply:
Doesn't feel very clean to remove the sln(x) files. But if people are ok with it, I am happy to leave it as to avoid generating even more test assets unnecessarily